posydon/popsyn/synthetic_population.py

```diff
@@ -392,7 +410,24 @@ def get_dco_merger_efficiency(self):
             efficiencies.append(eff)
             print(f'DCO merger efficiency at Z={met:1.2E}: {eff:1.2E} Msun^-1')
         self.met_merger_efficiency = np.array(self.met_merger_efficiency)
-        self.merger_efficiency = np.array(efficiencies)
+        self.merger_efficiency = {}
+        self.merger_efficiency['total'] = np.array(efficiencies)
+        # if the channel column is present compute the merger efficiency per channel
+        if "channel" in self.df_synthetic:
+            channels = np.unique(self.df_synthetic['channel'])
+            for ch in channels:
+                efficiencies = []
+                for met in self.met_merger_efficiency:
+                    sel = (self.df_synthetic['metallicity'] == met) & (self.df_synthetic['channel'] == ch)
+                    count = self.df_synthetic[sel].shape[0]
+                    if count > 0.:
+                        underlying_stellar_mass = self.df_synthetic.loc[sel,'underlying_mass_for_met'].values[0]
+                        eff = count/underlying_stellar_mass
+                    else:
+                        eff = np.nan
```

Comment:
This is not set to `0` or a negative number because this value is plotted in log scale, and at the moment, we have residual channels that might have discontinuous values and did not want to display sharp spikes. I am closing the issue for the moment and will address in the future when these values are used in an active science project.